sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -39,9 +39,16 @@ object StringUtils extends Logging {
    * throw an [[AnalysisException]].
    *
    * @param pattern the SQL pattern to convert
+   * @param escapeStr the escape string contains one character.
    * @return the equivalent Java regular expression of the pattern
    */
-  def escapeLikeRegex(pattern: String): String = {
+  def escapeLikeRegex(pattern: String, escapeStr: String): String = {
+    assert(escapeStr.length <= 1)
+    val escapeChar = if (escapeStr.length == 0) {
+      ""
+    } else {
+      escapeStr.charAt(0)
+    }
```

Comment:
Can you move this logic to `AstBuilder`?

Reply:
@maropu OK. I will move it.